### DISCUSSION TITLE
utilities doesn't override attribute selector

### DISCUSSION THREAD

**Discussion by kyimoemin** [tailwindlabs/tailwindcss#5974] [Help] [answered]
**What version of Tailwind CSS are you using?**

npm:@tailwindcss/postcss7-compat@^2.2.17

**What build tool (or framework if it abstracts the build tool) are you using?**

 postcss-cli ^7.0.39, CRA 4.0.3, webpack 4.2.2, @craco/craco ^6.4.0

**What version of Node.js are you using?**

v12.14.1

**What browser are you using?**

 Chrome

**What operating system are you using?**

Windows

**Reproduction URL**

https://codesandbox.io/s/modest-sara-n58c8?file=/src/App.js

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways. Unfortunately we can't provide support without a reproduction, and your issue will be closed with no comment if this is not provided.

**Describe your issue**

tailwind utilities classes do not override base layer styles if the element is selected by attribute. But it works without attribute selector. e.g.,

``` css
 @layer base {
  input[type="text"]{
   @apply rounded;
 }
}
```
```html
 <input type="text" class="rounded-none"/>
```

**Comment by adamwathan** [comment, marked as answer]
This is just how specificity in CSS works. `input[type="text"]` has a specificity of `011` and `.rounded-none` has a specificity of `010`.

![image](https://user-images.githubusercontent.com/4323180/140037123-ca0b8fbf-5ca1-4a13-9c8b-40e33cce9e9b.png)

So you'll need to either reduce the specificity of your base selector (that's what I'd do, just make it `[type="text"]`, you don't need the `input` part), or use Tailwind's `important` option to increase the specificity of all the utilities.
